internal/iam/role.go

```diff
@@ -327,3 +327,37 @@ func (p *projectRole) Actions() map[string]action.Type {
 	ret[action.SetPrincipals.String()] = action.SetPrincipals
 	return ret
 }
+
+// getRoleScopeId returns scopeID for the Role from the base type iam_role table
+// use this to get scope ID to determine which of the role subtype tables to operate on
+func getRoleScopeId(ctx context.Context, r db.Reader, roleId string) (string, error) {
+	const (
+		op    = "iam.getRoleScopeId"
+		query = `select scope_id from iam_role where public_id = ?`
```

Comment:
I think this should live in `internal/iam/query.go` based on our current pattern.

There are some plaintext queries that live outside `query.go` but they seem to only be used in special cases/helper functions (`Db.Now`, Db.hasTable, `InitCommand.verifyOplogIsEmpty`)